### PR TITLE
Fix: volume controller constantly update the ScheduledCondition

### PR DESCRIPTION
**Pull request by PhanLe1010**
When a volume is created with number of replicas more than available nodes and `AllowVolumeCreationWithDegradedAvailability` is enabled, volume controller constantly updates the volume's `ScheduledCondition`. This prevents other components to update the volume's spec since there will be a lot of conflicts. As a result, volume sometimes takes a long time to attach.

This PR makes sure volume controller only update volume's condition if the condition's status has changed

Longhorn/longhorn#2073

### CODE REVIEW

**Comment by PhanLe1010** [comment]
Notice that the `Condition` struct contains the time `LastTransitionTime`. Therefore, when the old code updates the VolumeConditionTypeScheduled, even if update the  VolumeConditionTypeScheduled to the same value, the Volume object still consider as has changed since the VolumeConditionTypeScheduled's LastTransitionTime has changed. This is a bit tricky.

File: controller/volume_controller.go [hunk 1]
```diff
@@ -430,6 +432,15 @@ func (vc *VolumeController) syncVolume(key string) (err error) {
 	return nil
 }
 
+// handleConditionLastTransitionTime rollback to the existing condition object if condition's value hasn't changed
+func handleConditionLastTransitionTime(existingStatus, newStatus *types.VolumeStatus) {
+	for conditionType, condition := range newStatus.Conditions {
+		if existingCondition, ok := existingStatus.Conditions[conditionType]; ok && existingCondition.Status == condition.Status {
```
Comment (yasker) [inline controller/volume_controller.go:438]: Also need to check the reason and message to ensure they're the same one.

Comment (PhanLe1010) [reply controller/volume_controller.go]: Thank you! Fixed

**Review by yasker — APPROVED** [review]
LGTM.

**Comment by yasker** [comment]
Is this ready to be merged?

**Comment by PhanLe1010** [comment]
Running E2E core tests but I think that it is safe to merge since the fix is simple

**Pull request merged**